src/common/Pagination/index.js

```diff
@@ -1,70 +1,157 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import { selectMovies } from '../../features/movies/moviesSlice';
+import { selectPopularPeople } from '../../features/people/popularPeopleSlice';
 import Button from "./Button";
 import Icon from "./Icon";
 import PageInfo from './PageInfo';
 import { Footer } from './styled';
 
 const Pagination = ({ type }) => {
+
+  const movies = useSelector(selectMovies);
+  const people = useSelector(selectPopularPeople);
+  const params = useParams();
+
+  const [page, setPage] = useState(+params.page);
+  const [backButtonState, setBackButtonState] = useState(false);
+  const [forwardButtonState, setForwardButtonState] = useState(false);
+
+  const backButtonStateHandler = () => {
+    if (type === "movies") {
+      if (page === 1) {
+        setBackButtonState(true);
+      } else
+        setBackButtonState(false);
+    };
+    if (type === "people") {
+      if (page === 1) {
+        setBackButtonState(true);
+      } else
+        setBackButtonState(false);
+    };
+  };
```

Comment:
O ile tą parametryzację pod spodem rozumiem (bo może być różna ilość total pages w zależności czy będziemy mieli do czynienia z people czy z movies), o tyle tutaj jest to chyba trochę przekombinowane, bo niezależnie czy typ będzie movies czy people warunek jest taki sam to po pierwsze. Druga sprawa opis tej zmiennej stanu dla mnie trochę niezrozumiały, bo jeśli mamy backButtonState jako true to wtedy jest przycisk disabled, co trochę miesza w moim mniemaniu. Można by tutaj to zamienić np. na backButtonDisactive, bo trzeba to będzie wykorzystywać później do atrybutu disabled, więc jeśli backButtonDisactive byłby na true, to i disabled również.
Jednak ja bym tutaj zrezygnował całkowicie z tej funkcji i do disabled przykazywał po prostu disabled={page===1} ;)

Reply:
Tak, można by trochę przerobić tą funkcję i to zrobię bo faktycznie w tym przypadku nie ma sensu rozdzielać na typy. Chyba już za późno było :D 
Co do samego użycia stanu zamiast reguły wprost wpisanej w komponent to kierowałem się tym, że jeśli coś się powtarza i jest wielokrotnie używane to powinno być zapisane w zmiennej.
A samą jej nazwę spróbuje jakoś dopracować :) 
Dzięki za uwagi.